pennylane/pauli/grouping/group_observables.py

```diff
@@ -59,38 +68,48 @@ class PauliGroupingStrategy:  # pylint: disable=too-many-instance-attributes
             the Pauli words, can be ``'qwc'`` (qubit-wise commuting), ``'commuting'``, or
             ``'anticommuting'``.
         graph_colourer (str): the heuristic algorithm to employ for graph
-            colouring, can be ``'lf'`` (Largest First) or ``'rlf'`` (Recursive
-            Largest First)
+                colouring, can be ``'lf'`` (Largest First), ``'rlf'`` (Recursive
+                Largest First), `dsatur` (DSATUR), or `gis` (IndependentSet). Defaults to ``'lf'``.
+
+    See Also:
+        `rustworkx.ColoringStrategy <https://www.rustworkx.org/apiref/rustworkx.ColoringStrategy.html#coloringstrategy>`_
+        for more information on the ``('lf', 'dsatur', 'gis')`` strategies.
 
     Raises:
-        ValueError: if arguments specified for ``grouping_type`` or
-            ``graph_colourer`` are not recognized
+        ValueError: if arguments specified for ``grouping_type`` or ``graph_colourer``
+        are not recognized.
     """
 
-    def __init__(self, observables, grouping_type="qwc", graph_colourer="rlf"):
+    def __init__(self, observables, grouping_type="qwc", graph_colourer="lf"):
         if grouping_type.lower() not in GROUPING_TYPES:
             raise ValueError(
                 f"Grouping type must be one of: {GROUPING_TYPES}, instead got {grouping_type}."
             )
 
-        self.grouping_type = grouping_type.lower()
-
         if graph_colourer.lower() not in GRAPH_COLOURING_METHODS:
             raise ValueError(
-                f"Graph colouring method must be one of: {list(GRAPH_COLOURING_METHODS)}, "
+                f"Graph colouring method must be one of: {GRAPH_COLOURING_METHODS}, "
                 f"instead got {graph_colourer}."
             )
 
-        self.graph_colourer = GRAPH_COLOURING_METHODS[graph_colourer.lower()]
+        self.graph_colourer = graph_colourer.lower()
+        self.grouping_type = grouping_type.lower()
         self.observables = observables
         self._wire_map = None
-        self._n_qubits = None
-        self.binary_observables = None
-        self.adj_matrix = None
-        self.grouped_paulis = None
+
+    @cached_property
+    def binary_observables(self):
```

Comment:
Is this just to add caching? Could we just add some caching to `binary_repr`? Just thinking about keeping the class design concise.

Reply:
In a standard workflow, will we ever hit this a second time? If we only ever calculate it once for a `PauliGroupingStrategy` instance, it probably adds more overhead than it's worth.

Reply:
So it is used both in the generation of the adjacency matrix and passed to the rlf function. So in theory it could be used a second time (although rlf is not the default anymore). I made it a cached property because that is what the original code was doing basically (just in a more rudimentary way):
https://github.com/PennyLaneAI/pennylane/blob/c7bedd8170bc1e5ad7ca4189e218dda3346c97ba/pennylane/pauli/grouping/group_observables.py#L129

I can also just erase completely this property and leave only `binary_repr`. I was leaving it for now to not change the structure of this class that much.

Reply:
👍